dio/lib/src/adapters/io_adapter.dart

```diff
@@ -187,27 +187,24 @@ class IOHttpClientAdapter implements HttpClientAdapter {
     Future<void>? cancelFuture,
     Duration? connectionTimeout,
   ) {
-    HttpClient client = onHttpClientCreate?.call(HttpClient()) ?? HttpClient();
     if (cancelFuture != null) {
+      final HttpClient client =
+          onHttpClientCreate?.call(HttpClient()) ?? HttpClient();
       client.userAgent = null;
       client.idleTimeout = Duration(seconds: 0);
       cancelFuture.whenComplete(() => client.close(force: true));
       return client..connectionTimeout = connectionTimeout;
     }
-    if (_defaultHttpClient == null) {
-      client.idleTimeout = Duration(seconds: 3);
-      if (onHttpClientCreate?.call(client) != null) {
-        client = onHttpClientCreate!(client)!;
-      }
-      client.connectionTimeout = connectionTimeout;
-      _defaultHttpClient = client;
-    }
-    return _defaultHttpClient!..connectionTimeout = connectionTimeout;
+    _cachedHttpClient ??=
+        onHttpClientCreate?.call(HttpClient()) ?? HttpClient();
+    return _cachedHttpClient!
+      ..idleTimeout = Duration(seconds: 3)
```

Comment:
The previous code only updates the `idleTimeout` when the first set. Is the new change intentional?

Reply:
No, which we also have another issue to expose the setting.